@SamuelTrahanNOAA the unit test has failed on my side. Would you please let me know what goes wrong?

[Jian.Kuang@hfe08 unittests]$ sh run_tests.sh
test_HydraIMPI_big (test_HydraIMPI.TestHydraIMPI) ... ok
test_HydraIMPI_max_OMP_NUM_THREADS (test_HydraIMPI.TestHydraIMPI) ... ok
test_HydraIMPI_max_notMPI (test_HydraIMPI.TestHydraIMPI) ... ok
test_HydraIMPI_max_ppn (test_HydraIMPI.TestHydraIMPI) ... ok
test_ShellCommand (test_ShellCommand.TestShellCommand) ... INFO:crow:file1: write mode wt
INFO:crow:file2: write mode wt
INFO:crow:Popen ['/bin/sh', '-c', 'cat $FILE1 $FILE2']
ok
test_AprunCrayMPI_big (test_SrunMPI.TestAprunCrayMPI) ... INFO:root:assertions not set yet
ok
test_AprunCrayMPI_max_ppn (test_SrunMPI.TestAprunCrayMPI) ... INFO:root:assertions not set yet
ok
test_ecFlowSuite (unittest.loader._FailedTest) ... ERROR
test_not_working_ec (test_ecFlowTaskarray.TestEcFlowTaskarray) ... INFO:crow.config:doc.suite: no overrides requested.
INFO:to_ecflow:populate job graph...
INFO:to_ecflow:201801011800: populate job graph...
INFO:to_ecflow:201801020000: populate job graph...
INFO:to_ecflow:201801020600: populate job graph...
INFO:to_ecflow:201801021200: populate job graph...
INFO:to_ecflow:201801021800: populate job graph...
INFO:to_ecflow:simplify job graph...
INFO:to_ecflow:201801011800: make suite definition in memory...
INFO:to_ecflow:201801011800: make ecf files in memory...
INFO:to_ecflow:201801020000: make suite definition in memory...
INFO:to_ecflow:201801020000: make ecf files in memory...
INFO:to_ecflow:201801020600: make suite definition in memory...
INFO:to_ecflow:201801020600: make ecf files in memory...
INFO:to_ecflow:201801021200: make suite definition in memory...
INFO:to_ecflow:201801021200: make ecf files in memory...
INFO:to_ecflow:201801021800: make suite definition in memory...
ERROR
test_FirstMax (test_exampleConfig.TestExampleConfig) ... ERROR
test_FirstTrue (test_exampleConfig.TestExampleConfig) ... ERROR
test_LastTrue (test_exampleConfig.TestExampleConfig) ... ERROR
test_NoneTrue (test_exampleConfig.TestExampleConfig) ... ERROR
test_bool_array (test_exampleConfig.TestExampleConfig) ... ERROR
test_calclist (test_exampleConfig.TestExampleConfig) ... ERROR
test_conditionals_on_empty_list (test_exampleConfig.TestExampleConfig) ... ERROR
test_default (test_exampleConfig.TestExampleConfig) ... ERROR
test_error_clause (test_exampleConfig.TestExampleConfig) ... ERROR
test_fcst_values (test_exampleConfig.TestExampleConfig) ... ERROR
test_file_namelist (test_exampleConfig.TestExampleConfig) ... ERROR
test_inherit (test_exampleConfig.TestExampleConfig) ... ERROR
test_inline_namelist (test_exampleConfig.TestExampleConfig) ... ERROR
test_int_array (test_exampleConfig.TestExampleConfig) ... ERROR
test_msg_clause (test_exampleConfig.TestExampleConfig) ... ERROR
test_not_working (test_exampleConfig.TestExampleConfig) ... ERROR
test_ordered_dict (test_exampleConfig.TestExampleConfig) ... ERROR
test_plus (test_exampleConfig.TestExampleConfig) ... ERROR
test_set (test_exampleConfig.TestExampleConfig) ... ERROR
test_string_array (test_exampleConfig.TestExampleConfig) ... ERROR
test_string_expansion (test_exampleConfig.TestExampleConfig) ... ERROR
test_strlen_func (test_exampleConfig.TestExampleConfig) ... ERROR
test_time_values (test_exampleConfig.TestExampleConfig) ... ERROR
test_has_threads (test_jobs.TestBoth) ... ok
test_is_mpi (test_jobs.TestBoth) ... ok
test_is_pure_serial (test_jobs.TestBoth) ... ok
test_num_ranks (test_jobs.TestBoth) ... ok
test_openmp_false (test_jobs.TestBoth) ... ok
test_openmp_true (test_jobs.TestBoth) ... ok
test_pure_openMP (test_jobs.TestBoth) ... ok
test_pure_serial (test_jobs.TestBoth) ... ok
test_spec_length (test_jobs.TestBoth) ... ok
test_has_threads (test_jobs.TestOpenMP) ... ok
test_individual_spec_is_not_mpi (test_jobs.TestOpenMP) ... ok
test_individual_spec_is_not_pure_serial (test_jobs.TestOpenMP) ... ok
test_individual_spec_is_openmp (test_jobs.TestOpenMP) ... ok
test_is_not_pure_serial (test_jobs.TestOpenMP) ... ok
test_is_pure_openmp (test_jobs.TestOpenMP) ... ok
test_total_ranks (test_jobs.TestOpenMP) ... ok
test_has_no_threads (test_jobs.TestSerial) ... ok
test_individual_spec_is_not_mpi (test_jobs.TestSerial) ... ok
test_individual_spec_is_not_openmp (test_jobs.TestSerial) ... ok
test_individual_spec_is_pure_serial (test_jobs.TestSerial) ... ok
test_is_not_pure_openmp (test_jobs.TestSerial) ... ok
test_is_pure_serial (test_jobs.TestSerial) ... ok
test_total_ranks (test_jobs.TestSerial) ... ok
test_comp_and (test_simplify.TestSimplify) ... ok
test_comp_nand (test_simplify.TestSimplify) ... ok
test_comp_or (test_simplify.TestSimplify) ... ok
test_simp_a_and_not_a (test_simplify.TestSimplify) ... ok
test_simp_a_or_not_a (test_simplify.TestSimplify) ... ok
test_simp_extended_expr (test_simplify.TestSimplify) ... ok
test_simp_gobbledygook (test_simplify.TestSimplify) ... ok
test_simp_not_not_a_or_not_b (test_simplify.TestSimplify) ... ok
test_10_109 (test_util.TestRankstoNodes) ... ok
test_10_3 (test_util.TestRankstoNodes) ... ok
test_24_31 (test_util.TestRankstoNodes) ... ok
test_24_62 (test_util.TestRankstoNodes) ... ok
test_3_10 (test_util.TestRankstoNodes) ... ok
test_obj_match (test_yaml_io.TestYamlIO) ... ok
test_text_match (test_yaml_io.TestYamlIO) ... ok

======================================================================
ERROR: test_ecFlowSuite (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_ecFlowSuite
Traceback (most recent call last):
  File "/apps/intel/intelpython3/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/apps/intel/intelpython3/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_ecFlowSuite.py", line 11, in <module>
    conf=from_file('ecftest.yaml')
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 57, in from_file
    with open(file,'rt') as fopen:
FileNotFoundError: [Errno 2] No such file or directory: 'ecftest.yaml'


======================================================================
ERROR: test_not_working_ec (test_ecFlowTaskarray.TestEcFlowTaskarray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_ecFlowTaskarray.py", line 24, in setUp
    fd.write(self.suite_defs[defname])
TypeError: tuple indices must be integers or slices, not str

======================================================================
ERROR: test_FirstMax (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_FirstTrue (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_LastTrue (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_NoneTrue (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_bool_array (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_calclist (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_conditionals_on_empty_list (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_default (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_error_clause (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_fcst_values (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_file_namelist (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_inherit (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_inline_namelist (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_int_array (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_msg_clause (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_not_working (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_ordered_dict (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_plus (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_set (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_string_array (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_string_expansion (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_strlen_func (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

======================================================================
ERROR: test_time_values (test_exampleConfig.TestExampleConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/tests/unittests/test_exampleConfig.py", line 17, in setUp
    crow.config.validate(self.config.fcst)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/__init__.py", line 94, in validate
    obj._validate(stage)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/eval_tools.py", line 347, in _validate
    tmpl._check_scope(self,stage,memo)
  File "/scratch1/NCEPDEV/global/Jian.Kuang/crow-merge/workflow/CROW/crow/config/template.py", line 82, in _check_scope
    if self.__my_id in memo:
TypeError: argument of type 'NoneType' is not iterable

----------------------------------------------------------------------
Ran 70 tests in 1.322s

FAILED (errors=25)

--- a/crow/config/eval_tools.py
+++ b/crow/config/eval_tools.py
@@ -304,7 +304,7 @@ class dict_eval(MutableMapping):
     def __iter__(self):
         for k in self.__child.keys(): yield k
     def _inherit(self,stage,memo=None):
-
+        """!If requested, inherit contents of other scopes into this scope."""
         # Make sure we don't get infinite recursion:
         if memo is None: memo=set()
         if id(self) in memo:
@@ -322,8 +322,10 @@ class dict_eval(MutableMapping):
                 _logger.warning(f'{self._path}: Inherit is not an !Inherit.  Error?')
         elif superdebug:
             _logger.debug(f'{self._path}: no Inherit')
+
     def _validate(self,stage,memo=None):
         """!Validates this dict_eval using its embedded Template object, if present """
+        if memo is None: memo=set()
         if self.__is_validated: return
         self.__is_validated=True
 

